File: frontend/src/services/jobs.ts
import api from "../lib/api";

export type JobRunStatus = "running" | "ok" | "failed";

export interface JobLastRun {
  status: JobRunStatus;
  summary: string;
  findings: number;
  started_at: string;
  finished_at: string | null;
}

// Matches models.JobStatus — one registered background job (internal/jobs)
// and its current on/off state and most recent run, if any.
export interface JobStatus {
  name: string;
  description: string;
  schedule: string;
  enabled: boolean;
  last_run: JobLastRun | null;
}

export interface RunJobResult {
  summary: string;
  findings: number;
}

export const jobsApi = {
  list: () => api.get<JobStatus[]>("/jobs").then((r) => r.data),

  setEnabled: (name: string, enabled: boolean) =>
    api.put(`/jobs/${name}/enabled`, { enabled }).then((r) => r.data),

  runNow: (name: string) => api.post<RunJobResult>(`/jobs/${name}/run`).then((r) => r.data),
};

File: backend/internal/models/jobs.go
package models

import "time"

// JobLastRun is the most recent execution of a background job, for the
// admin Automation panel.
type JobLastRun struct {
	Status     string     `json:"status"`
	Summary    string     `json:"summary"`
	Findings   int32      `json:"findings"`
	StartedAt  time.Time  `json:"started_at"`
	FinishedAt *time.Time `json:"finished_at,omitempty"`
}

// JobStatus is one registered background job's static definition plus its
// current on/off state and last run, if any.
type JobStatus struct {
	Name        string      `json:"name"`
	Description string      `json:"description"`
	Schedule    string      `json:"schedule"`
	Enabled     bool        `json:"enabled"`
	LastRun     *JobLastRun `json:"last_run,omitempty"`
}

type SetJobEnabledRequest struct {
	Enabled bool `json:"enabled"`
}
